tools/distpartitioning/utils.py

```diff
@@ -420,3 +422,26 @@ def get_idranges(names, counts):
 
     return tid_dict, gid_dict
 
+
+def mem_snapshot(tag, rank):
+    """
+    Utility function to take a snapshot of the usage of system resources
+    at a given point of time.
+    
+    Parameters: 
+    -----------
+    tag : string
+        string provided by the user for bookmarking purposes
+    rank : integer
+        process id of the participating process
+    """
+    GB = 1024 * 1024 * 1024
+    MB = 1024 * 1024
+
+    mem = psutil.virtual_memory()
+    avail = mem.available / MB
+    used = mem.used / MB
+    total = mem.total / MB
+
+    mem_string = '{:.0f} (MB) total, {:.0f} (MB) used, {:.0f} (MB) avail'.format(total, used, avail)
```

Comment:
Try f-string to increase the readability?
https://realpython.com/python-string-formatting/#3-string-interpolation-f-strings-python-36

Reply:
Done as suggested.